fixtures/dev-env/tests/index.test.ts

```diff
@@ -649,3 +649,42 @@ describe("startDevWorker: ProxyController", () => {
 		await expect(res.text()).resolves.toBe("UserWorker:3");
 	});
 });
+
+describe("regressions", () => {
+	// https://github.com/cloudflare/workers-sdk/issues/5095
+	test("#5095 requests where body (of >= 2 bytes) is not drained sometimes fail", async () => {
+		const run = await fakeStartUserWorker({
+			script: `
+				export default {
+					async fetch(request) {
+						return new Response("UserWorker:1");
+					}
+				}
+			`,
+		});
+
+		const COUNT = 30;
+		const STEP_DELAY = 4; // setting to less than 4 results in errorCount === 0 indicating race condition or a request influencing a subsequent request
+		let requestCount = 0;
+		let errorCount = 0;
+
+		await Promise.allSettled(
+			Array.from({ length: COUNT }).map(async (_, i) => {
+				await new Promise((r) => setTimeout(r, i * STEP_DELAY));
+
+				return undici
+					.fetch(run.proxyWorkerUrl, {
+						method: "POST",
+						body: new Uint8Array(2), // 2 bytes
```

Comment:
Why 2 bytes?

Reply:
Any smaller and the bug does not appear.

Reply:
This bit was actually in the test name 😄